packages/core/src/environment.ts

```diff
@@ -1 +1,4 @@
-export const IS_PRODUCTION = process.env.NODE_ENV === 'production';
+export const IS_PRODUCTION =
+  typeof process !== 'undefined'
+    ? process.env.NODE_ENV === 'production'
+    : false;
```

Comment:
Based on the end goal described [here](https://github.com/statelyai/xstate/pull/2820#issuecomment-970406584) and that this would minify better I think we should switch to this:
```suggestion
export const IS_PRODUCTION =
  typeof process !== 'undefined'
    ? process.env.NODE_ENV === 'production'
    : true;
```